sdk/storage/inc/common/storage_credential.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+#include <mutex>
+
```

Comment:
These should all be in the Azure::Storage::Credentials namespace

Reply:
According to the guidelines [here](https://azure.github.io/azure-sdk/cpp_design.html#namespaces),  the namespace should be `Azure::<Group>::<Service>`. So I think `Azure::Storage::Blobs` is fine, but `Credentials` is not a service, I don't feel it's right.

I took reference from .NET sdk, they put these credential definitions into namespace `azure::storage`, which I think is good. Because we don't need to add namespace prefix when using them like `Credentials::SharedKeyCredential`